STIPS/STIP-002.md

```diff
@@ -0,0 +1,150 @@
+- aToken.balanceOf() returns scaledBalance * liquidityIndex
+- aToken.transfer(quantity) updates the scaledBalance by adding (quantity/liquidityIndex) to it.
+    - liquidityIndex = (timeDiff/secondsPerYear)*interestRate + 1, hence liquidityIndex is always > 1
+    - timeDiff = block.timestamp - lastUpdatedTimestamp
+    - lastUpdatedTimestamp = timestamp of last block in which there was an interaction with this aToken specific reserve, interactions include calling deposit, withdraw, repay, borrow, flashloan and swapBorrowRate functions on the reserve.
+
+Now in our ExplicitERC20#transferFrom function,
+- We call existingBalance = aToken.balanceOf(setToken), which sets 
+existingBalance = initialScaledBalance * index
+- By calling aToken.transfer(quantity), we update baseBalance to newScaledBalance
+newScaledBalance = initialScaledBalance + (quantity/index)
+- Then we set newBalance = aToken.balanceOf(setToken)
+    - newBalance = newScaledBalance * index
+- Finally we require, newBalance == existingBalance + quantity
+    - LHS = newBalance
+        - = newScaledBalance * index
+        - = (initialScaledBalance + (quantity/index)) * index
+        - = (initialScaledBalance * index) + (quantity/index)*index
+    - RHS = existingBalance + quantity
+        - = (initialScaledBalance * index) + quantity
+- Cancelling, initialScaledBalance * index from LHS and RHS,
+- We essentially require, (quantity/index) * index == quantity, with index > 1
+- Which will not always be true because solidity uint256 has limited precision and there is a chance of `(quantity/index) * index` to be greater/lesser than `quantity` by 1 wei.
+
+## Feasibility Analysis
+
+### Option 1:
+Instead of over-optimizing our checks to test for both undercollaterlization and overcollaterlization. We can just have checks to prevent undercollaterlization. This would prevent reverting in cases when `newBalance` of aToken is greater than `existingBalance + quantity` by 1 wei, and thus decrease the failure rate of issuance/redemption (maybe to 33%, cause we can consider less than, equal to, and greater than to be 3 equal cases). This option is the best considering time. The change is contained within the smart contracts and doesn't bubble up the stack leading to bigger changes in our backend.
+
+
+
+## Open Questions
+
+1. Do we restrict usage of these modified issuance modules (which are linked to the new libraries) from third-party use?
+    - Why did we have the strict checks in the first place?
+        - To make sure the set is not undercollateralized 
+            - Eg. To stop tokens which collects fees on transfer from being a component in a SetToken
+        - This strict check achieved the above requirement, but at the same time it also prevented overcollateralization
```

Comment:
This is not exactly true, over-collateralization happens all the time as a _result_ of manager actions (or fee accruals). However, this would not allow over-collateralization based on issuance.

Reply:
Yups, I meant during issuance/redemption.